src/components/editor_flexbox.tsx

```diff
@@ -1,84 +1,149 @@
+import SelectWrapper from "./select_wrapper";
+import StringInput from "./string_input";
+import StringInputWrapper from "./string_input_wrapper";
+import Tab from "react-bootstrap/Tab";
+import TableWrapper from "./table_wrapper";
+import Tabs from "react-bootstrap/Tabs";
+import TextInput from "./text_input";
+import TextInputWrapper from "./text_input_wrapper";
+import USStateSelectWrapper from "./us_state_select_wrapper";
 
+const components = {
+  Collapsible,
+  DateInput: DateInputWrapper,
+  Figure: FigureWrapper,
+  NumberInput: NumberInputWrapper,
+  Photo: PhotoWrapper,
+  PhotoInput: PhotoInputWrapper,
+  PrintSection,
+  Select: SelectWrapper,
+  StringInput: StringInputWrapper,
+  Table: TableWrapper,
+  TextInput: TextInputWrapper,
+  USStateSelect: USStateSelectWrapper,
+  DateStr: DateStr,
+  Tab: Tab,
+  Tabs: Tabs,
+};
 
+let initialTemplateText = `
+<br></br>
+<PrintSection label="Print Report">
+<Collapsible header="Collapsible — Instructions">
+    Here is an example of a collapsible component
+    1. First item
+    2. Second item
+    3. Third item
+</Collapsible>
+<Figure src="/images/hpwh/hpwh_components.jpeg">
+  The components in a common HPWH (BASC 2015)
+</Figure>
+<PhotoInput id="photo_input" label="Photo Input">
+  Provide a photo
+</PhotoInput>
+<NumberInput label="Number Input" min={0} path="number_input.value"  />
+Value:{props.doc.number_input?.value}
+<StringInput label="String Input" path="string_input.value" />
+Value:{props.doc.string_input?.value}
+</PrintSection>
+<USStateSelect label="State" path="state_select.value" />
+Value:{props.doc.state_select?.value}
+`;
 
-const EditorFlexBox = () => {
+function generateTemplateView(templateText: string) {
+  let MDXContent;
+  try {
+    MDXContent = evaluateSync(templateText, {
+      ...provider,
+      Fragment: _Fragment,
+      jsx: _jsx,
+      jsxs: _jsxs,
+      useMDXComponents,
+      useDynamicImport: true,
+    });
+  } catch {
+    throw new Error("Error evaluating MDX");
+  }
+  return MDXContent;
+}
 
-  const [text, setText] = useState('');
+const EditorFlexBox: FC = () => {
+  let savedTemplateText = localStorage.getItem("templateText");
+  if (savedTemplateText) {
+    initialTemplateText = savedTemplateText;
+  }
 
-  const handleChange = (event: ChangeEvent<HTMLTextAreaElement>) => {
-    setText(event.target.value);
-    event.target.style.height = 'auto'; // Reset the height to auto to adjust the size
-    event.target.style.height = `${event.target.scrollHeight}px`; // Set the height to the scrollHeight to adjust the size
+  const [templateText, setTemplateText] = useState(initialTemplateText);
+  const [hasError, setHasError] = useState(false);
+  const [mdxComponent, setMdxComponent] = useState(generateTemplateView(initialTemplateText));
+  const handleButtonClick = () => {
+    try {
+      setMdxComponent(generateTemplateView(templateText));
+      localStorage.setItem("templateText", templateText);
```

Comment:
How about creating a function `handleTemplateTextChange()` that is called whenever the template text changes. This method calls `setTemplateText()` and executes `localStorage.setItem("templateText", templateText)`. That way the localStorage is updated on every keystroke.

Then there is no need to update localStorage here.

With the ErrorBondary suggestion I make below, the try/catch here can be eliminated.